packages/models/src/lib/plugin-config-runner.ts

```diff
@@ -24,3 +24,15 @@ export const runnerConfigSchema = z.object(
 );
 
 export type RunnerConfig = z.infer<typeof runnerConfigSchema>;
+
+export const esmObserver = z.object({
+  next: z.function().args(z.unknown()).returns(z.void()).optional(),
+});
+export type EsmObserver = z.infer<typeof esmObserver>;
+
+export const esmRunnerConfigSchema = z
+  .function()
+  .args(esmObserver.optional())
```

Comment:
What is the purpose of the Observer argument? As a Code PushUp maintainer, I don't understand what I should use it for. Do I have to use it? If so, why is it optional? I would guess to a user it would make even less sense.

Reply:
refactored the observer part